src/main/java/fi/aalto/cs/apluscourses/model/ModuleInstallerImpl.java

```diff
@@ -0,0 +1,143 @@
+package fi.aalto.cs.apluscourses.model;
+
+import fi.aalto.cs.apluscourses.utils.async.TaskManager;
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import org.jetbrains.annotations.NotNull;
+
+public class ModuleInstallerImpl<T> implements ModuleInstaller {
```

Comment:
could the IJ's [`ModuleManager`](https://github.com/JetBrains/intellij-community/blob/002b9b884ce17d5894eff6b00e9affb89f324ee0/platform/projectModel-impl/src/com/intellij/openapi/module/impl/ModuleManagerImpl.java) be of any help here?

Reply:
ModuleManager is finally called: see `IntelliJModule.Loader.load()` but before that we must download (fetch) the zip, unzip, read dependencies from IML file (and start their installations). So *loading* module to is just a small part of installation.

Reply:
Please note that I just changed this (outer) class to fix a bug (it became also simpler). I push that now if you don't mind.